hoss/dimension_utilities.py

```diff
@@ -422,22 +559,48 @@ def add_index_range(
 
     """
```

Comment:
The ordering and the shape are things I need to get from varinfo. I dont have that information. Updated DAS-2241 for this.

Reply:
Maybe rephrasing a different comment: I'm very uncomfortable with the thought of merging code with known problems into `main`. Going back to one of the things mentioned in the TRT breakout last week:

> Teams ought to have a high-quality, maintainable baseline without known defects (particularly in new features) that is deployable to the production environment at all times.

Instead of making a massive PR that does 80% of a lot of stuff, this probably should be submitted piecemeal, with each piece being made 100% solid in a series of smaller PRs.

If you and David feel we're at a point that we can't break the changes down in that way, then a compromise might be to update this PR to merge into a feature branch. Then once all of the multiple PRs are merged into the feature branch, the feature branch can be merged into `main`.

Reply:
I think terminology has confused some of this discussion.  The question of reversing is not one of reversing dimensions, but the lat/lon variables coming from the Coordinates attribute.  The recommendation here is that the get_coordinates method itself should return in a standard order (reversed in case), based upon the variable name - and not using reverse as shown here.  

I don't think this is a case of "Known Issue".  

We are planning to release a version for NSIDC to start testing with, which may not handle the 3D cases or other issues, but this release should not break any existing use cases, and should not truly break, but simply not handle as desired.  Incremental feature release is a tenet of agile development.

Reply:
Changes are in commit 756f7c0

I have simplified the method. The order for SMAP is 'projected_y', 'projected_x'. The override section of the code is only used by SMAP at the moment. It can be generalized if we can get that order of dimensions from varinfo. I am not sure if the order of dimensions is used for other products currently handled by hoss.

Reply:
The use of sets for required_dimensions is based on how it is returned by varinfo and how it was originally in hoss before my changes. the bounds update requires the dimensions to be a set , It fails for a list

Reply:
There's a difference between a set of all dimensions for all required variables (as used in the prefetch function), which aggregates all `Variable.dimensions` attributes (which individually are lists), and the dimensions on an individual variable. `Variable.dimensions` is not a set for ordering purposes, [it is a list](https://github.com/nasa/earthdata-varinfo/blob/main/varinfo/variable.py#L243).

With regards to `bounds` - we know that the pseudo-dimensions won't have a `bounds` attribute, so you might be better to not try to find bounds variables for them. Then you'll avoid any compatibility issues there.

Reply:
https://github.com/nasa/harmony-opendap-subsetter/blob/681b20d15c611e7cddb0ec904ded0038c63db45c/hoss/dimension_utilities.py#L100

   ```
 required_dimensions_nc4 = get_opendap_nc4(
        opendap_url, required_dimensions, output_dir, logger, access_token, config
    )
```
In the above method required_dimensions is a set as well

Reply:
The add_index_range is a lot simpler now